pkg/dns/treecache.go

```diff
@@ -130,32 +90,38 @@ func (cache *TreeCache) GetSubCache(path ...string) *TreeCache {
 	return childCache
 }
 
-func (cache *TreeCache) SetSubCache(key string, subCache *TreeCache, path ...string) {
-	cache.m.Lock()
-	defer cache.m.Unlock()
+func (cache *TreeCache) setSubCache(key string, subCache *TreeCache, path ...string) {
```

Comment:
I'd rather do the locking outside of this function and document that this assumes it is called with the lock held.